File: app/MyAgent/insecure/vuln_demo.py
"""
INTENTIONALLY VULNERABLE — SAST/CodeQL demo fixtures (KPI #15).

DO NOT DEPLOY. These endpoints take untrusted HTTP input (a recognized
*remote* taint source for CodeQL) and feed it straight into dangerous sinks so
the CodeQL default query suite raises real code-scanning alerts:

  * py/command-line-injection
  * py/code-injection
  * py/sql-injection
  * py/path-injection
"""
import os
import sqlite3
import subprocess

from flask import Flask, request

app = Flask(__name__)


@app.route("/ping")
def ping():
    # py/command-line-injection: remote input -> shell
    host = request.args.get("host", "")
    return os.popen("ping -c 1 " + host).read()


@app.route("/run")
def run():
    # py/command-line-injection: remote input -> subprocess with shell=True
    cmd = request.args.get("cmd", "")
    return subprocess.check_output(cmd, shell=True)


@app.route("/calc")
def calc():
    # py/code-injection: remote input -> eval
    expr = request.args.get("expr", "0")
    return str(eval(expr))


@app.route("/user")
def user():
    # py/sql-injection: remote input concatenated into a query
    name = request.args.get("name", "")
    conn = sqlite3.connect("app.db")
    cur = conn.cursor()
    cur.execute("SELECT * FROM users WHERE name = '" + name + "'")
    return str(cur.fetchall())


@app.route("/read")
def read():
    # py/path-injection: remote input -> filesystem path
    name = request.args.get("file", "")
    with open("/var/data/" + name) as fh:
        return fh.read()


if __name__ == "__main__":
    app.run()

File: app/insecure/vuln_demo.js
/*
 * INTENTIONALLY VULNERABLE — SAST/CodeQL demo fixtures (KPI #15).
 *
 * DO NOT DEPLOY. These routes take untrusted HTTP input (a recognized *remote*
 * taint source for CodeQL) and feed it into dangerous sinks so the CodeQL
 * default query suite raises real code-scanning alerts:
 *
 *   - js/command-line-injection
 *   - js/code-injection
 *   - js/path-injection
 */
const express = require("express");
const cp = require("child_process");
const fs = require("fs");

const app = express();

app.get("/ping", (req, res) => {
  // js/command-line-injection: remote input -> shell
  const host = req.query.host;
  cp.exec("ping -c 1 " + host, (err, stdout) => res.send(stdout));
});

app.get("/calc", (req, res) => {
  // js/code-injection: remote input -> eval
  const expr = req.query.expr;
  res.send(String(eval(expr)));
});

app.get("/read", (req, res) => {
  // js/path-injection: remote input -> filesystem path
  const name = req.query.file;
  fs.readFile("/var/data/" + name, "utf8", (err, data) => res.send(data));
});

app.listen(3000);
